lib/grpc/client/adapters/mint.ex

```diff
@@ -0,0 +1,182 @@
+defmodule GRPC.Client.Adapters.Mint do
+  @moduledoc """
+  A client adapter using mint
+  """
+
+  alias GRPC.{Channel, Credential}
+  alias GRPC.Client.Adapters.Mint.{ConnectionProcess, StreamResponseProcess}
+
+  @behaviour GRPC.Client.Adapter
+
+  @default_connect_opts [protocols: [:http2]]
+  @default_transport_opts [timeout: :infinity]
+
+  @impl true
+  def connect(%{host: host, port: port} = channel, opts \\ []) do
+    opts = Keyword.merge(@default_connect_opts, connect_opts(channel, opts))
+
+    channel
+    |> mint_scheme()
+    |> ConnectionProcess.start_link(host, port, opts)
+    |> case do
+      {:ok, pid} -> {:ok, %{channel | adapter_payload: %{conn_pid: pid}}}
+      # TODO add proper error handling
+      _error -> {:ok, %{channel | adapter_payload: %{conn_pid: nil}}}
+    end
+  end
+
+  @impl true
+  def disconnect(%{adapter_payload: %{conn_pid: pid}} = channel)
+      when is_pid(pid) do
+    ConnectionProcess.disconnect(pid)
+    {:ok, %{channel | adapter_payload: %{conn_pid: nil}}}
```

Comment:
This throws away the previous adapter payload instead of just setting the conn_pid to nil. Is this intentional? Why not just set the adapter_payload to nil directly?

Reply:
Yeah! that makes sense